systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaTopicScalabilityUtils.java

```diff
@@ -0,0 +1,127 @@
+public class KafkaTopicScalabilityUtils {
+    private static final Logger LOGGER = LogManager.getLogger(KafkaTopicUtils.class);
+    protected static ResourceManager resourceManager = ResourceManager.getInstance();
+
+    public static void createTopicsViaK8s(ExtensionContext extensionContext, String clusterName, String topicPrefix,
+                                          int numberOfTopics, int numberOfPartitions, int numberOfReplicas, int minInSyncReplicas) {
+        LOGGER.info("Creating topics via Kubernetes");
+
+        for (int i = 0; i < numberOfTopics; i++) {
+            String currentTopicName = topicPrefix + i;
+            LOGGER.debug("Creating topic {}", currentTopicName);
+
+            CompletableFuture.runAsync(() ->
+                    resourceManager.createResource(extensionContext, false, KafkaTopicTemplates.topic(
+                            clusterName, currentTopicName, numberOfPartitions, numberOfReplicas, minInSyncReplicas, INFRA_NAMESPACE).build()));
+        }
+    }
+
+    public static void checkTopicsState(String topicPrefix, int numberOfTopics, Enum<?> state) {
+        List<CompletableFuture<?>> topics = new ArrayList<>();
+
+        for (int i = 0; i < numberOfTopics; i++) {
+            String currentTopic = topicPrefix + i;
+            topics.add(CompletableFuture.runAsync(() ->
+                    KafkaTopicUtils.waitForKafkaTopicStatus(INFRA_NAMESPACE, currentTopic, state)
+            ));
+        }
+
+        CompletableFuture<Void> allTopics = CompletableFuture.allOf(topics.toArray(new CompletableFuture[0]))
+                .thenRun(() -> LOGGER.info("All topics are in correct state"));
+
+        allTopics.join();
+    }
+
+    public static void checkTopicsNotReady(String topicPrefix, int numberOfTopics) {
+        LOGGER.info("Verifying that topics are in NotReady state");
+        KafkaTopicScalabilityUtils.checkTopicsState(topicPrefix, numberOfTopics, CustomResourceStatus.NotReady);
+    }
+
+    public static void checkTopicsReady(String topicPrefix, int numberOfTopics) {
+        LOGGER.info("Verifying that topics are in Ready state");
+        KafkaTopicScalabilityUtils.checkTopicsState(topicPrefix, numberOfTopics, CustomResourceStatus.Ready);
+    }
+
+    public static void checkTopicsContainConfig(String topicPrefix, int numberOfTopics, Map<String, Object> config) {
+        LOGGER.info("Verifying that topics contain right config");
+        List<CompletableFuture<?>> topics = new ArrayList<>();
+
+        for (int i = 0; i < numberOfTopics; i++) {
+            String currentTopic = topicPrefix + i;
+            topics.add(CompletableFuture.runAsync(() -> {
+                try {
+                    String json = cmdKubeClient(kubeClient().getNamespace()).exec("get", "kafkatopic", currentTopic, "-o", "jsonpath='{.spec.config}'").out();
+                    HashMap<?, ?> mapping = new ObjectMapper().readValue(json.replaceAll("'", ""), HashMap.class);
```

Comment:
Wouldn't `KafkaTopicResource.kafkaTopicClient().inNamespace(namespaceName).withName(topicName).get().getSpec().getConfig()` be better than this? You will get in correct format and you don't need to map it.